qiskit_ibm_runtime/options/options.py

```diff
@@ -123,48 +124,78 @@ def _set_if_exists(name: str, _inputs: dict, _options: dict) -> None:
             "coupling_map": coupling_map,
             "basis_gates": sim_options.get("basis_gates", Unset),
         }
 
         for fld in [
-            "resilience_level",
+            "default_precision",
+            "default_shots",
+            "seed_estimator",
+            "dynamical_decoupling",
             "resilience",
             "twirling",
-            "dynamical_decoupling",
-            "seed_estimator",
         ]:
-            _set_if_exists(fld, inputs, options_copy)
+            _set_if_exists(fld, output_options, options_copy)
+
+        resilience_options = output_options.get("resilience", {})
+        # Convert zne/pec settings to layer mitigation
+        layer_mitigation = []
+        if resilience_options.pop("zne_mitigation", None):
+            layer_mitigation.append("zne")
+        if resilience_options.pop("pec_mitigation", None):
+            layer_mitigation.append("pec")
+        if layer_mitigation:
+            resilience_options["layer_mitigation"] = layer_mitigation
+            output_options["resilience"] = resilience_options
+
+        # rename meas_num_randomizations and meas_shots_per_randomization
+        meas_learning_options = resilience_options.get("measure_noise_learning", {})
+        if "meas_num_randomizations" in meas_learning_options:
+            meas_learning_options["num_randomizations"] = meas_learning_options.pop(
+                "meas_num_randomizations"
+            )
+        if "meas_shots_per_randomization" in meas_learning_options:
+            meas_learning_options["shots_per_randomization"] = meas_learning_options.pop(
+                "meas_shots_per_randomization"
+            )
 
-        inputs["execution"] = options_copy.get("execution", {})
-        inputs["execution"].update(
+        output_options["execution"] = options_copy.get("execution", {})
+        output_options["execution"].update(
             {
                 "noise_model": sim_options.get("noise_model", Unset),
                 "seed_simulator": sim_options.get("seed_simulator", Unset),
             }
         )
 
         # Add arbitrary experimental options
-        if isinstance(options_copy.get("experimental", None), dict):
-            inputs = merge_options(inputs, options_copy.get("experimental"))
+        experimental = options_copy.get("experimental", None)
+        if isinstance(experimental, dict):
+            new_keys = {}
+            for key in list(experimental.keys()):
+                if key not in output_options:
+                    new_keys[key] = experimental.pop(key)
+            output_options = merge_options(output_options, experimental)
+            if new_keys:
+                output_options["experimental"] = new_keys
 
         # Remove image
-        inputs.pop("image", None)
+        output_options.pop("image", None)
 
-        inputs["version"] = OptionsV2._VERSION
-        _remove_dict_unset_values(inputs)
+        output_options["version"] = OptionsV2._VERSION
```

Comment:
Should the version go in "inputs" instead? I know the options get flattened elsewhere, so maybe it doesn't make a difference, but if we switch that, I believe the version will be in the wrong spot.

Reply:
Am I also wrong here? Please resolve if so.

Reply:
You were right. It's fixed in [7e49516](https://github.com/Qiskit/qiskit-ibm-runtime/pull/1460/commits/7e49516094b89c40f7cb69ca8541a6e9b51c7e5f)